### PR TITLE
Disabled items breaking focus in menu

### PR DESCRIPTION
**Pull request by roginfarrer**
**What**:

Address issue here: https://github.com/paypal/downshift/issues/557

Since no event handlers are passed down to disabled items in the menu, clicking on disabled items will redirect focus, which we avoid with the `onMouseDown` event handler. This will pass down this event.

**Why**:

If disabled items are kept in the list, clicking on them will redirect focus, an undesirable behavior.

**How**:

In the eventHandlers object, we already have an `onMouseDown` `event.preventDefault()` to prevent the focus change. 

**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Ready to be merged
      
- [x] Added myself to contributors table

### CODE REVIEW

File: src/downshift.js [hunk 1]
```diff
@@ -850,7 +850,9 @@ class Downshift extends Component {
       }),
     }
 
-    const eventHandlers = rest.disabled ? {} : enabledEventHandlers
+    const eventHandlers = rest.disabled
+      ? {onMouseDown: enabledEventHandlers.onMouseDown}
```
Comment (kentcdodds) [inline src/downshift.js]: Let's add a code comment explaining why this one's getting special treatment.

**Comment by roginfarrer** [comment]
Sounds good! Added a comment. Thanks!

**Comment by kentcdodds** [comment]
Thanks so much for your help! I've added you as a collaborator on the project. Please make sure that you review the `other/MAINTAINING.md` and `CONTRIBUTING.md` files (specifically the bit about the commit messages and the git hooks) and familiarize yourself with the code of conduct (we're using [the contributor covenant](http://contributor-covenant.org/)). You might also want to [watch the repo](https://help.github.com/articles/watching-repositories/) to be notified when someone files an issue/PR. Please continue to make PRs as you feel the need (you can make your branches directly on the repo rather than your fork if you want). Thanks! And welcome to the team :)

**Pull request merged**